parlai/scripts/train_model.py

```diff
@@ -220,17 +235,26 @@ def __init__(self, opt):
 
     def validate(self):
         opt = self.opt
+
+        if self.valid_world is None:
+            # we need to load the world now
+            self.valid_world = _maybe_load_world(self.agent, opt, 'valid')
+
         # run evaluation on valid set
-        valid_report, self.valid_world = run_eval(
-            self.agent, opt, 'valid', opt['validation_max_exs'],
-            valid_world=self.valid_world)
+        valid_report = sync_object(run_eval(
+            self.valid_world, opt, 'valid', opt['validation_max_exs'],
+        ))
 
         # logging
-        if opt['tensorboard_log'] is True:
+        if opt['tensorboard_log'] is True and is_primary_worker():
             self.writer.add_metrics('valid', int(
```

Comment:
casting as an int already takes the floor, you don't need that call

Reply:
(y)